https://bugs.gentoo.org/877793

otherwise some rust crates will 404 due to wrong path being used in the mirror url by emerge/wget

Index: /var/tmp/portage/sys-apps/portage-3.0.38.1/work/portage-3.0.38.1/lib/portage/package/ebuild/fetch.py
===================================================================
--- portage-3.0.38.1/lib/portage/package/ebuild/fetch.py
+++ portage-3.0.38.1/lib/portage/package/ebuild/fetch.py
@@ -1094,9 +1094,15 @@ def fetch(
                 writemsg(_("Invalid mirror definition in SRC_URI:\n"), noiselevel=-1)
                 writemsg("  %s\n" % (myuri), noiselevel=-1)
         else:
-            if (restrict_fetch and not override_fetch) or force_mirror:
-                # Only fetch from specific mirrors is allowed.
-                continue
+            writemsg("!!! lefetch: '%s' -> '%s' " % (myuri, myfile))
+            #!!! lefetch: 'https://distfiles.gentoo.org/distfiles/layout.conf' -> '.layout.conf.distfiles.gentoo.org'
+            #saved as /var/cache/distfiles/.layout.conf.distfiles.gentoo.org
+            if myfile[:13] != '.layout.conf.':
+                if (restrict_fetch and not override_fetch) or force_mirror:
+                    # Only fetch from specific mirrors is allowed.
+                    writemsg("!! disallowed by FEATURES='force-mirror'\n")
+                    continue
+            writemsg("\n")
             primaryuris = primaryuri_dict.get(myfile)
             if primaryuris is None:
                 primaryuris = []
